### DISCUSSION TITLE
ESP32 serial miner

### DISCUSSION THREAD

**Discussion by joaquinbvw** [revoxhere/duino-coin#438] [General]
Hi! As I told some Discord users I have an ESP32 serial miner working good. Here is the code. I will try to improve it using two cores since the WiFi is not in use anymore. To have this running you will need to match the baudrate between the Arduino code and the AVR miner python code, it can be 115200 or 500000 maximum (ESP32 doesn't like higher baudrates). Also you will have to copy uniqueID.h and uniqueID.cpp from the Arduino_Code folder to your sketch folder. I've forced the hashrate to be lower too, feel free to change it to whatever value you want it's a delay on line #69, I just did this because a high hashrate results in the server sending higher difficulty jobs and it will give you less coins.

I've been having some problems with the WiFi version, not really sure if it's my internet provider or the code itself. I will work on it too but my involvement will be slower now.
[DuinoCoinESP32Serial.txt](https://github.com/revoxhere/duino-coin/files/6372337/DuinoCoinESP32Serial.txt)

**Comment by revoxhere** [comment]
Nice job!

**Comment by BastelPichi** [comment]
Cool! Will check it out tomorrow

**Comment by LordKlaus** [comment]
I've tried to upload the code on an ESP32. I've taken the uniqueID.h from the AVR folder, but I get this error while attempting the code upload:
_sketch\DuinoCoinESP32Serial.ino.cpp.o:(.literal._Z5setupv+0x1c): undefined reference to `_UniqueID'
collect2.exe: error: ld returned 1 exit status
exit status 1_

**Reply by joaquinbvw** [reply]
Are you using the Arduino IDE? I have worked just with the Arduino IDE, don't know about that collect2.exe...

By the way I forgor to mention that I have changed the baudrate to 500000, if you want to use that baudrate you will need to change the AVR miner python code. You can change it back to 115200 too, I don't see much difference in performance actually.

**Reply by LordKlaus** [reply]
Yes Arduino IDE 1.8.13. Tried also to set the speed to 115200, but I get the same error that I've never seen before.

**Reply by LordKlaus** [reply]
In any case, seems that the real problem is "undefined reference to `_UniqueID'" located in file uniqueID.h
